### ISSUE TITLE
Nginx config broken on "modern" systems

### ISSUE THREAD

**Issue by aleho** [nextcloud/notify_push#47]
At least on Debian and Fedora (but I suspect on many others as well) the default configuration out of the box is resolving `localhost` to `::1` instead of `127.0.0.1`.

This is the reason why nginx will error with
```
connect() failed (111: Connection refused) while connecting to upstream, client: 127.0.0.1, server: my.cloud, request: "GET /push//test/mapping/1 HTTP/1.1", upstream: "http://[::1]:7867/test/mapping/1", host: "my.cloud"
```
when doing a self-test, because notify push will expect IPv4 (in the setup example).

I suggest being explicit about the host in the example, by replacing `localhost` with `127.0.0.1`.

**Comment by alerque** [comment]
And by "modern", your probably mean anything with IPv6 enabled, even if it isn't the default for internet facing networking....

Just to note though, this doesn't seem to affect by Apache based proxy the way it does you. My localhost resolves to `::1` but I didn't have any trouble on that front getting everything setup. The only real hurdle I had was getting the `trusted_proxies` setup right with the public IPv4 address of my proxy. As such are you sure it is even an issue with the notify app and not with Nginx itself?

**Comment by aleho** [comment]
On all my systems localhost nowadays resolves to ::1, so I guess Nginx is correct here, they're just using the system resolver.

I don't think it's an issue with notify_push, but in a default setup the steps from the README.md will not result in a working setup. So my recommendation is to be more specific and change the docs to say

```nginx
location /push/ {
    proxy_pass http://127.0.0.1:7867/;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
}
```
which, with the seemingly default behavior of notify_push to bind to IPv4 only, will result in a working configuration.

This is a bug in my opinion that is more severe than it might look like at first glance, because it doesn't break immediately and even the self-test command succeeds, with only the Nginx error log showing an actual failure.

**Comment by chrisingenhaag** [comment]
Hi @aleho ,
I had the same issue. I strongly recommend to use the new possibility to listen on a unix socket instead of a port. My proxy_pass statement looks like the following and works well.

`
proxy_pass http://unix:/var/run/notify_push.socket:/;
`

**Comment by aleho** [comment]
This has been fixed with 9f67831bea96140344c74b19ce042528eabde63f.
